src/Splunk.ModularInputs/Splunk/ModularInputs/EventWriter.cs

```diff
@@ -94,11 +94,17 @@ public class EventWriter
         {
             if (completed)
                 throw new ObjectDisposedException("EventWriter already disposed.");
+
+            // Don't start the eventQueueMonitor up until we actually want to write
+            // events so we don't get empty <stream/> elements in cases where
+            // we don't actually queue anything.
             if (eventQueueMonitor == null)
-                // Don't start the eventQueueMonitor up until we actually want to write
-                // events so we don't get empty <stream/> elements in cases where
-                // we don't actually queue anything.
-                eventQueueMonitor = Task.Run(() => WriteEventsFromQueue());
+                lock (synchronizationObject)
+                {
+                    if (eventQueueMonitor == null)
```

Comment:
Yes it is. If we have 2 threads that are going to write something, they can find uninitialized eventQueueMonitor variable on the first method call and try to take the lock. The first who takes the lock completes initialization and releases the lock, the second one should make additional check to skip unnecessary initialization.